pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -540,15 +540,17 @@ func mongodbContainer(version string, volumeMounts []corev1.VolumeMount) contain
 	mongoDbCommand := []string{
 		"/bin/sh",
 		"-c",
-		// we execute the pre-stop hook once the mongod has been gracefully shut down by the agent.
-		`while [ ! -f /data/automation-mongod.conf ]; do sleep 3 ; done ; sleep 2 ;
-# start mongod with this configuration
-mongod -f /data/automation-mongod.conf ;
+		`
+# run post-start hook to handle version changes.
+# during a version change, mongod will be stopped and the
+# container will be restarted, at which points this hook runs.
+/hooks/version-upgrade-hook
```

Comment:
We could have used the container lifecycle post-start hook instead of including the hook in our script. This would cause the hook to run concurrently with `mongod`, resulting in the server starting up one extra time before being killed again.

We opted to trigger the post-start hook from inside the bash script to avoid this extra restart.

Reply:
indeed

Reply:
we can probably just name it without hook since it's in the hooks directory

Reply:
That's true, renamed!